lib/inferno/dsl/assertions.rb

```diff
@@ -197,6 +197,58 @@ def no_content_type_message
       def bad_content_type_message(expected, received)
         "Expected `Content-Type` to be `#{expected}`, but found `#{received}`"
       end
+
+      # Check that all Must Support elements defined on the given profile are present in the given resources.
+      # Must Support elements are identified on the profile StructureDefinition and pre-parsed into metadata,
+      # which may be customized prior to the check by passing a block. Alternate metadata may be provided directly.
+      # Set test suite config flag debug_must_support_metadata: true to log the metadata to a file for debugging.
+      #
+      # @param resources [Array<FHIR::Resource>]
+      # @param profile_url [String]
+      # @param validator_name [Symbol] Name of the FHIR Validator that references the IG the profile is in
+      # @param metadata [Hash] MustSupport Metadata (optional),
+      #        if provided the check will use this instead of re-generating metadata from the profile
+      # @param requirement_extension [String] Extension URL that implies "required" as an alternative to the MS flag
+      # @yield [Metadata] Customize the metadata before running the test
+      # @return [void]
+      def assert_must_support_elements_present(resources, profile_url, validator_name: :default, metadata: nil,
+                                               requirement_extension: nil, &)
+        missing_elements = missing_must_support_elements(resources, profile_url, validator_name:, metadata:,
+                                                                                 requirement_extension:, &)
+        assert missing_elements.empty?, missing_must_support_elements_message(missing_elements, resources)
+      end
+
+      # Find any Must Support elements defined on the given profile that are missing in the given resources.
+      # Must Support elements are identified on the profile StructureDefinition and pre-parsed into metadata,
+      # which may be customized prior to the check by passing a block. Alternate metadata may be provided directly.
+      # Set test suite config flag debug_must_support_metadata: true to log the metadata to a file for debugging.
+      #
+      # @param resources [Array<FHIR::Resource>]
+      # @param profile_url [String]
+      # @param validator_name [Symbol] Name of the FHIR Validator that references the IG the profile is in
+      # @param metadata [Hash] MustSupport Metadata (optional),
+      #        if provided the check will use this instead of re-generating metadata from the profile
+      # @param requirement_extension [String] Extension URL that implies "required" as an alternative to the MS flag
+      # @yield [Metadata] Customize the metadata before running the test
+      # @return [Array<String>] List of missing elements
+      def missing_must_support_elements(resources, profile_url, validator_name: :default, metadata: nil,
```

Comment:
I realize this method still doesn't really fit here in assertions, but I'm not sure where else to put it. I don't want to include the `MustSupportTest` automatically in every test, and I don't feel like it's super worth creating a whole new module just for one method. I'm open to suggestions

Reply:
Ok this `missing_must_support_elements` has been migrated to the new `MustSupportAssessment` module, and nearly all the logic from that is encapsulated in a class with just a few public entrypoints based on what's here and in the US Core test kit